### PR TITLE
Jsonschema example

### PR DESCRIPTION
**Pull request by wanghaoran1988**

### CODE REVIEW

File: hack/metadata.schema.json [hunk 1]
```diff
@@ -1,114 +1,735 @@
...
+                                                                            "pods"
+                                                                        ]
+                                                                    }
+                                                                ]
+                                                            }
+                                                        },
+                                                        "resourceNames": {
+                                                            "$id": "#/properties/rbac/properties/roles/items/anyOf/0/properties/rules/items/anyOf/0/properties/resourceNames",
+                                                            "type": "array",
+                                                            "title": "The resourceNames schema",
```
Comment (qhua948) [inline hack/metadata.schema.json:428]: Need to make this optional

Comment (wanghaoran1988) [reply hack/metadata.schema.json:428]: this schema is auto generated, and I made no changes to that, this is just for testing, @qhua948 Could you update the existing one that already in the repo?

Comment (qhua948) [reply hack/metadata.schema.json:428]: :+1:

File: hack/app_sre_pr_check.sh [hunk 2]
```diff
@@ -1,7 +1,32 @@
...
+do
+   echo "convert $f to json"
+   if ! python3 hack/yamltojson.py $f; then
+     echo "convert failed"
+     exit 1
+   else
+     echo "convert succeed"
+   fi
+done
+#Verify the metadata.json are valide
```
Comment (qhua948) [inline hack/app_sre_pr_check.sh:19]: s/valide/valid

File: hack/app_sre_pr_check.sh [hunk 3]
```diff
@@ -1,7 +1,32 @@
...
+jfiles=$(find . -name 'metadata.json')
+for f in $jfiles
+do
+   echo "validating the jsonschema for $f"
+   if ! $CONTAINER_ENGINE run --rm -v $(pwd):/json quay.io/haowang/jsonschema:latest -i /json/$f /json/hack/metadata.schema.json; then
```
Comment (qhua948) [inline hack/app_sre_pr_check.sh:24]: can we invoke a single container and then run a `find ... exec ...` in there? I think it's better than invoking docker each time

File: hack/app_sre_pr_check.sh [hunk 4]
```diff
@@ -1,7 +1,32 @@
...
+     echo "validating failed: $f"
```
Comment (qhua948) [inline hack/app_sre_pr_check.sh:25]: s/validating/validation

File: hack/app_sre_pr_check.sh (hunk 3 above)
Comment (wanghaoran1988) [reply hack/app_sre_pr_check.sh:24]: This is more easy for me ,we can fail at anytime when the validation failed.

**Pull request merged**